javascript/diff_match_patch_uncompressed.js

```diff
@@ -2397,16 +2386,22 @@ diff_match_patch.patch_obj.prototype.toString = function() {
 
 // CLOSURE:begin_strip
 // Lines below here will not be included in the Closure-compatible library.
-
-// Export these global variables so that they survive Google's JS compiler.
-// In a browser, 'this' will be 'window'.
-// Users of node.js should 'require' the uncompressed version since Google's
-// JS compiler may break the following exports for non-browser environments.
-/** @suppress {globalThis} */
-this['diff_match_patch'] = diff_match_patch;
-/** @suppress {globalThis} */
-this['DIFF_DELETE'] = DIFF_DELETE;
-/** @suppress {globalThis} */
-this['DIFF_INSERT'] = DIFF_INSERT;
-/** @suppress {globalThis} */
-this['DIFF_EQUAL'] = DIFF_EQUAL;
+if (typeof module === 'object') {
+  // The following export code was added by @ForbesLindesay
```

Comment:
who is @ForbesLindesay and why did they add this code?
I think this would be better suited as a `git` commit or a `Co-authored-by:` trailer on the commit, given that it's trying to attribute the change but the comment doesn't indicate why it's there

Reply:
The original developer of the repository these changes came from, scroll down to 2015
https://github.com/JackuB/diff-match-patch/commits/master/

Reply:
What if we eliminated all of this and instead used only module syntax?

```js
export default diff_match_patch;
export diff_match_path;
export DIFF_DELETE;
export DIFF_INSERT;
export DIFF_EQUAL;
```

then we omit `index.mjs` and instead, generate a Common-JS artifact? or not. maybe it's too late, but I don't even know if I care about Common-JS, if we're breaking free in this repo. won't any project with a recent build _have_ to support ESM? I was thinking that several major libraries are now exporting their packages untranspiled

Reply:
There are plenty of CJS projects around, it's still the NodeJS default, and using `module.exports` here alongside the `mjs` file (using with the references in `package.json` to differentiate them) means it can work for both.

I'm happy to make this ESM-only, if that's what you want - I certainly only need ESM.